src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalMessagesOpt.java

```diff
@@ -43,40 +43,37 @@ public JSONObject getComponentMessages() {
     
     @Override
     public void getComponentMessages(MessageCacheItem cacheItem) {
-        Locale bestMatch = Locale.lookup(Arrays.asList(new Locale.LanguageRange((dto.getLocale()))),
-        		getSupportedLocales());
 		try {
-			String filePath = FormatUtils.format(OFFLINE_RESOURCE_PATH, dto.getComponent(), bestMatch.toLanguageTag());
-			Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl(), filePath);
-			
-			URI uri = Thread.currentThread().getContextClassLoader().
-					getResource(path.toString()).toURI();
-			
-			Map<String, String> messages = null;
-	    	if (uri.getScheme().equals("jar")) {
-				try(FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
-					path = fileSystem.getPath(path.toString());
-					messages = JSONBundleUtil.getMessages(path);
-				}
-			} else {
-				path = Paths.get(uri);
-				messages = JSONBundleUtil.getMessages(path);
-			}
+			InputStream is = getInputStream();
+			JSONParser jsonParser = new JSONParser();
+			JSONObject jsonObject = (JSONObject) jsonParser.parse(new InputStreamReader(is, "UTF-8"));
+			Map<String, String> messages = (JSONObject) jsonObject.get("messages");
 			cacheItem.setCacheItem(messages, null, System.currentTimeMillis(), null);
 		} catch (Exception e) {
 			logger.debug(e.getMessage());
-			// Do not update cacheItem
 		}
     }
-    
-    private List<Locale> getSupportedLocales() {
-		ProductService ps = new ProductService(dto);
-		Set<String> supportedLanguages = ps.getSupportedLanguageTags(DataSourceEnum.Bundle);
-        logger.debug("supported languages: [{}]", supportedLanguages.toString());
-    	List<Locale> supportedLocales = new LinkedList<Locale>();
-    	for (String languageTag : supportedLanguages) {
-    		supportedLocales.add(Locale.forLanguageTag(languageTag));
-    	}
-    	return supportedLocales;
-    }
+
+	private InputStream getInputStream() {
+		String locale = LocaleUtility.fmtToMappedLocale(dto.getLocale()).toLanguageTag();
+		while (true) {
+			String filePath = FormatUtils.format(OFFLINE_RESOURCE_PATH, dto.getComponent(), locale);
+			Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl(), filePath);
+			InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream(path.toString());
```

Comment:
Why change FileSystem to getResourceAsStream?

Reply:
Because FileSystems.newFileSyustem fails in nested jars, but getResourceAsStream does not. So the problem is just in getting the list of components or supported locales (and only in Java 11 or below), not in actually reading the resource bundle.